src/pages/food/components/FoodDetailsPage.tsx

```diff
@@ -17,47 +19,119 @@ const TagList = styled.ScrollView`
-      ) }
-    >
-      <Box px="l">
-        <Subtitle mb="s">PRODUCE</Subtitle>
-        <Title mb="s">{titleHelper(name)}</Title>
-      </Box>
-      <TagList horizontal showsHorizontalScrollIndicator={ false }>
-        {tags && tags.map((tag: TagType) => (
-          <Box key={ tag.id } alignSelf="flex-start" mr="xs">
-            <Tag value={ tag.name } />
-          </Box>
-        ))}
-      </TagList>
-    </DrawerCard>
-  </Box>
-);
+      />
+      <DrawerCard topOffset={ 300 }>
+        <Box px="l" mt="s">
+          <Subtitle mb="s">PRODUCE</Subtitle>
+          <Title mb="s">{titleHelper(name)}</Title>
+        </Box>
+        <TagList horizontal showsHorizontalScrollIndicator={ false }>
+          {tags && tags.map((tag: TagType) => (
+            <Box key={ tag.id } alignSelf="flex-start" mr="xs">
+              <Tag value={ tag.name } />
+            </Box>
+      ))}
+        </TagList>
+        <Grid>
+          <Column>
+            <NutritionChart progress={ carbsPercent / 100 } progressColor={ colors.neoncarrot } />
+            <Box alignItems="center">
+              <Box position="absolute" top="-65px">
+                <Title>
+                  {carbsPercent}
+                  %
+                </Title>
+              </Box>
+              <Text mt="s">CARBS</Text>
+            </Box>
+          </Column>
+          <Column>
+            <NutritionChart progress={ proteinPercent / 100 } progressColor={ colors.neoncarrot } />
+            <Box alignItems="center">
+              <Box position="absolute" top="-65px">
+                <Title>
+                  {proteinPercent}
+                  %
+                </Title>
+              </Box>
+              <Text mt="s">PROTEINS</Text>
+            </Box>
+          </Column>
+          <Column>
+            <NutritionChart progress={ fatPercent / 100 } progressColor={ colors.neoncarrot } />
+            <Box alignItems="center">
+              <Box position="absolute" top="-65px">
+                <Title>
+                  {fatPercent}
+                  %
+                </Title>
+              </Box>
+              <Text mt="s">FAT</Text>
+            </Box>
+          </Column>
+        </Grid>
+        <Box mt="xxl">
+          <Grid ml="xl" mr="xl">
+            <Column alignItems="flex-start">
+              {nutrition.map((item) => (
+                <Text key={ item.name } size="h1" mb="l">{item.name}</Text>
+          ))}
+            </Column>
+            <Column alignItems="flex-end">
+              {nutrition.map((item) => (
+                <Text key={ item.name } size="h1" mb="l">
+                  {item.value}
+                  g
+                </Text>
+          ))}
+            </Column>
+          </Grid>
+        </Box>
+      </DrawerCard>
+      <Grid position="absolute" left="0" right="0" bottom={ bottom > 0 ? bottom : 'xxxl' } px="l" justifyContent="center">
+        <Button variant="warning" title="Add to stock" />
```

Comment:
This button should use onPress = navigation.push('StockDetails', foodID)

Reply:
In other words, onPress = { onAdd }

Reply:
Done